### ISSUE TITLE
Using pdfkit

### ISSUE THREAD

**Issue by techtonik** [adambisek/string-pixel-width#1]
Maybe this will help to get width in different fonts. https://github.com/badges/shields/blob/master/lib/measure-text.js

**Comment by adambisek** [comment]
@techtonik Thanks, this looks good, but I am not sure, how about rendering speed. Do you have any experience with speed? It looks like [pdfkit](https://www.npmjs.com/package/pdfkit) is using its own rendering, that could be fast (faster than rendering into headless browser).

**Comment by techtonik** [comment]
No, I haven't measured performance. Was looking for alternative library without PDF in its name.

**Comment by aquelehugo** [comment]
You can use the library PDFKit uses internally: https://github.com/foliojs/fontkit

Code for measuring string width would be as follows:
```js
var fontkit = require('fontkit');

var font = fontkit.openSync('/path/to/font.ttf');

function getStringWidth(str, useKerning) {
    if (useKerning) {
        return font.layout(text).positions.reduce((widthSum, glyphPosition) => widthSum + glyphPosition.xAdvance, 0);
    }
    return font.glyphsForString(str).reduce((widthSum, glyph) => widthSum + glyph.advanceWidth, 0));
}
```

**Comment by adambisek** [comment]
Thanks for your advice. Will look at that.
But, one of the prerequisities when this package has been developed was speed. I am not sure that manipulating in PDF would be fast enough.

**Comment by aquelehugo** [comment]
It doesn't use PDF, it is the library for font loading only. I don't know how you would get the glyphs width in pixels exactly though, maybe you can get it from the pdfkit's implementation.
fontkit only provides width for glyphs in their original scale. That's enough for me because I just need the proportions.
Of course it will not be faster than your method which just access arrays, but there's a point you missed: kerning. Width for 'Vo' is not the same as the sum for 'V' and 'o' separately, for instance. The 'o' is rendered closer to the 'V' to improve readability.

**Comment by adambisek** [comment]
Wow, sounds very promising!
Will dive into when I have a little bit time!

**Comment by techtonik** [comment]
Make sure to checkout https://github.com/badges/shields/issues/1379 for performance discussion.
